src/Compilers/Core/Portable/SourceGeneration/RunResults.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis
+{
+    public readonly struct GeneratorDriverRunResult
+    {
+        internal GeneratorDriverRunResult(ImmutableArray<GeneratorRunResult> results, ImmutableArray<SyntaxTree> trees, ImmutableArray<Diagnostic> diagnostics)
+        {
+            this.Results = results;
+            this.SyntaxTrees = trees;
+            this.Diagnostics = diagnostics;
+        }
+
+        public ImmutableArray<GeneratorRunResult> Results { get; }
+
+        public ImmutableArray<Diagnostic> Diagnostics { get; }
+
+        public ImmutableArray<SyntaxTree> SyntaxTrees { get; }
```

Comment:
Are these the generated trees, the original trees + generated trees? Think the name is a bit ambiguous here. At the least we need a comment to describe what this is.

Reply:
Added doc comments to clarify (and because public API)

---
In reply to: [460214895](https://github.com/dotnet/roslyn/pull/46296#discussion_r460214895) [](ancestors = 460214895)